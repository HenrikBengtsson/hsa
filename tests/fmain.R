source("hsa/cstruct1.R")

outpath <- file.path("tests", ".checks")
dir.create(outpath, recursive = TRUE, showWarnings = FALSE)
outprefix <- file.path(outpath, "hap1_full_all_22_100000")
cat(sprintf("Prefix of output files: %s\n", sQuote(outprefix)))

# Input data
path <- file.path("inst", "exdata")
contacts <- file.path(path, "contact_hap1_full_all_22_100000.txt")

## Parameters
mak <- 1L
Iscovfile <- FALSE
K <- 1L
lscov0 <- 0L

data <- read.table(file = contacts, header = FALSE)
data <- as.matrix(data)
lsmap0 <- list(data)

set.seed(12345)

options(digits = 7L, scipen = 0L) ## Reproducible print() output

## AD HOC: Trick cstruct1.R code to write files with 12 digits
## (still plenty) instead of 15 digits for easier 'diff' comparisons
write.table <- function(x, ...) {
  for (kk in seq_along(x)) {
    if (is.double(x[[kk]])) x[[kk]] <- round(x[[kk]], digits = 12L)
  }
  utils::write.table(x, ...)
}

## Benchmarking history:
## commit 0880732: ~270 secs
## commit ccc77de: ~265 secs
dt <- system.time({
  out <- fmain(lsmap0 = lsmap0, lscov0 = lscov0, outfile = outprefix,
               Maxiter = 3L, submaxiter = 100L, lambda = 25, Leapfrog = 20,
               epslon = 0.003, mkfix = 0, rho = 0, mk = mak)
})
print(out)
cat("\n\n")

cat("Files produced:\n")
file1 <- sprintf("%s.txt", outprefix)
file2 <- sprintf("%sbeta.txt", outprefix)
cat(sprintf("- file 1: %s (%d bytes; %s)\n",
            sQuote(basename(file1)), file.size(file1), tools::md5sum(file1)))
cat(sprintf("- file 2: %s (%d bytes; %s)\n",
            sQuote(basename(file2)), file.size(file2), tools::md5sum(file2)))
cat("\n\n")

cat("\n\n")
print(sessionInfo())

cat("\nProcessing time:\n")
print(dt)

cat("\nValidation:\n")
data0_1 <- read.table(file.path(path, basename(file1)), header = FALSE)
data1_1 <- read.table(file1, header = FALSE)
stopifnot(all.equal(data1_1, data0_1))

data0_2 <- read.table(file.path(path, basename(file2)), header = FALSE)
data1_2 <- read.table(file2, header = FALSE)
stopifnot(all.equal(data1_2, data0_2))
