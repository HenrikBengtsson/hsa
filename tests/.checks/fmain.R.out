Prefix of output files: 'tests/.checks/hap1_full_all_22_100000'
-0.9811756 -1.814995 
LLK0:  185.8771 
number of nodes: 351 
0 :  -1.197186 -1.895176 
LLK after GLM:  231.111 
LLK before and after outlier removal: 231.111 232.614 ( 1.089996 -1.953391 ) 
LLK before and after smoothing: 232.614 222.2878 ( 0.8464303 -1.691769 ) 
0 0 0 1 0 0 0 1 0 0 0 1 1 0 0 0 1 0 0 0 1 232.614 
1 :  -1.264218 -1.979761 
LLK after GLM:  240.6074 
LLK before and after outlier removal: 240.6074 235.252 ( 0.9789422 -1.94245 ) 
LLK before and after smoothing: 240.6074 227.9489 ( 0.8092063 -1.662084 ) 
0 0 0 1 0 0 0 1 0 0 0 1 1 0 0 0 1 0 0 0 1 240.6074 
2 :  -1.307129 -1.996761 
LLK after GLM:  244.2109 
LLK before and after outlier removal: 244.2109 241.5977 ( 0.9589737 -1.967493 ) 
LLK before and after smoothing: 244.2109 232.2521 ( 0.8032123 -1.666825 ) 
0 0 0 1 0 0 0 1 0 0 0 1 1 0 0 0 1 0 0 0 1 244.2109 
[[1]]
     [,1] [,2] [,3]
[1,]    1    0    0
[2,]    0    1    0
[3,]    0    0    1

[[2]]
[1] 0 0 0

[[3]]
     [,1] [,2] [,3]
[1,]    1    0    0
[2,]    0    1    0
[3,]    0    0    1

[[4]]
               [,1]         [,2]          [,3]
  [1,] -1.014509778 -0.323567762  0.7042020356
  [2,] -0.774085735 -0.166177865  0.6280167108
  [3,] -0.847213419 -0.000360394  0.5048282458
  [4,] -0.758338979  0.153739806  0.3550729941
  [5,] -0.552227196  0.277741246  0.2452989103
  [6,] -0.453795068  0.572839995  0.0325998104
  [7,] -0.334094351  0.738888425 -0.0626315820
  [8,] -0.145251012  0.931229527 -0.1418510037
  [9,]  0.027653595  1.190168635 -0.3256382053
 [10,]  0.064228968  1.264476180 -0.3590147693
 [11,]  0.139443970  1.379468648 -0.2974083367
 [12,]  0.143011496  1.357301419 -0.2094323934
 [13,]  0.139267700  1.420372878 -0.1774713741
 [14,]  0.184518172  1.433366346 -0.1301534885
 [15,]  0.180765118  1.463843481 -0.0952894512
 [16,]  0.217007944  1.484343026 -0.0978215999
 [17,]  0.198817383  1.515533507 -0.0763194910
 [18,]  0.169826784  1.524562947 -0.1119419966
 [19,]  0.155661973  1.504689373 -0.0617399136
 [20,]  0.143274021  1.484222082 -0.0192137836
 [21,]  0.130643363  1.433838753 -0.0369274949
 [22,]  0.086736985  1.440348677 -0.0290479090
 [23,]  0.060394775  1.457878765 -0.0087411883
 [24,]  0.090562774  1.407279383  0.0010185395
 [25,]  0.073745337  1.392042954  0.0581137128
 [26,]  0.020207035  1.231670555  0.1522614199
 [27,]  0.089957084  1.088397989  0.1736695675
 [28,] -0.017470916  0.833572060  0.3617872147
 [29,]  0.012690224  0.891977437  0.3562127728
 [30,] -0.010304761  0.875115311  0.3168368469
 [31,]  0.016248391  0.865905542  0.2854422820
 [32,]  0.030089869  0.890089241  0.3022622668
 [33,]  0.068386052  0.885550336  0.2895407693
 [34,]  0.070146151  0.843184014  0.2863765484
 [35,]  0.063943746  0.833487501  0.2500324470
 [36,]  0.095699766  0.831404593  0.1963996329
 [37,]  0.060561174  0.846701887  0.1958197369
 [38,]  0.038843468  0.809277011  0.1959599350
 [39,]  0.030753313  0.788414845  0.1601247105
 [40,]  0.046695460  0.726319704  0.1508948622
 [41,]  0.010113185  0.717177192  0.1186513128
 [42,] -0.024964001  0.501765008 -0.2194173002
 [43,] -0.251742298  0.366122642 -0.3841005413
 [44,] -0.226586446  0.230465196 -0.5636400216
 [45,] -0.171638966  0.340719786 -0.5225113326
 [46,] -0.022095055  0.594364857 -0.0002300111
 [47,] -0.010777104  0.661710829  0.0542318702
 [48,] -0.046423683  0.645176842  0.0460691709
 [49,] -0.065866655  0.658295762  0.0053076275
 [50,] -0.105193697  0.636532747  0.0266325190
 [51,] -0.130571173  0.607747076  0.0092956747
 [52,] -0.149750490  0.653662019  0.0007182828
 [53,] -0.159820857  0.663044173 -0.0581888763
 [54,] -0.279746498  0.761004864 -0.3390941235
 [55,] -0.473917675  0.736593244 -0.3983281865
 [56,] -0.447971135  0.615736896 -0.1597945836
 [57,] -0.231903369  0.417747864  0.2781579781
 [58,] -0.112525385  0.484856036  0.2636941187
 [59,] -0.061010517  0.486120051  0.2403118396
 [60,] -0.018552405  0.473510965  0.2485646143
 [61,] -0.031135205  0.434212141  0.2313233243
 [62,]  0.020339023  0.434471320  0.2510250842
 [63,]  0.058004649  0.404598574  0.2463046193
 [64,]  0.051108495  0.369307533  0.2091690480
 [65,]  0.104225836  0.451627990  0.1984600386
 [66,]  0.114250255  0.366221149  0.2381080004
 [67,]  0.139916557  0.334006132  0.2136869461
 [68,]  0.128905003  0.292264128  0.1985018013
 [69,]  0.154473443  0.312094405  0.2643884690
 [70,]  0.088574578  0.305003377  0.2618377059
 [71,]  0.116178454  0.268477609  0.2549976906
 [72,]  0.109448999  0.249255744  0.3159773027
 [73,]  0.122769685  0.290690871  0.3387376081
 [74,]  0.089951827  0.239208463  0.3744396395
 [75,]  0.017786545 -0.132959488  0.4079485488
 [76,]  0.070161686 -0.168960178  0.3552555416
 [77,]  0.032436040 -0.192020980  0.3425098711
 [78,]  0.068464836 -0.233631346  0.3297860922
 [79,]  0.067879626 -0.260200373  0.2767683934
 [80,]  0.107433538 -0.264491206  0.2817663659
 [81,]  0.116312948 -0.257714399  0.2340796247
 [82,]  0.394886961 -0.487530524 -0.0535267314
 [83,]  0.109544047 -0.309939025  0.1398560905
 [84,]  0.076905248 -0.299899076  0.1163313628
 [85,]  0.055251546 -0.309516185  0.1588335394
 [86,]  0.006653781 -0.315605802  0.1456018893
 [87,]  0.002816193 -0.321511763  0.1015970352
 [88,] -0.039843201 -0.330514695  0.1164579933
 [89,] -0.335382987 -0.379492449  0.1291362870
 [90,] -0.299590273 -0.404679290  0.0943046238
 [91,] -0.329350922 -0.451062522  0.0805551529
 [92,] -0.353287621 -0.461662551  0.0187337380
 [93,] -0.367700741 -0.516317923  0.0451552096
 [94,] -0.317535604 -0.536582419  0.0173786278
 [95,] -0.267012082 -0.596909220 -0.0338022115
 [96,] -0.235833616 -0.568378505 -0.0877860170
 [97,] -0.212400095 -0.558593270 -0.1421024960
 [98,] -0.169574002 -0.620461153 -0.1127396104
 [99,] -0.175738287 -0.621419251 -0.1550376458
[100,] -0.134008769 -0.588992301 -0.1621497389
[101,] -0.096573599 -0.563346474 -0.1735888604
[102,] -0.088935311 -0.614309349 -0.1705282355
[103,] -0.090405047 -0.617580046 -0.2079646615
[104,] -0.073832309 -0.585153703 -0.2194768587
[105,] -0.049539008 -0.629551932 -0.2192835618
[106,] -0.037592924 -0.575464187 -0.1825370387
[107,] -0.007742361 -0.548712825 -0.2176908763
[108,]  0.006396971 -0.544017377 -0.2580815098
[109,]  0.024407278 -0.516683097 -0.2454731341
[110,]  0.086142305 -0.506361569 -0.2490136648
[111,]  0.086875254 -0.538849472 -0.2771623759
[112,]  0.118463380 -0.562340610 -0.2630009796
[113,]  0.086792354 -0.578287709 -0.2360078507
[114,]  0.120559989 -0.588010950 -0.1977133988
[115,]  0.126934607 -0.550030927 -0.2091214224
[116,]  0.099105598 -0.535283261 -0.1872496735
[117,]  0.132721713 -0.526637437 -0.1584549747
[118,]  0.108613850 -0.497489752 -0.1299111774
[119,]  0.108922430 -0.486343600 -0.1737243475
[120,]  0.142159589 -0.492379956 -0.1965337003
[121,]  0.142457078 -0.437008049 -0.2041103021
[122,]  0.145781594 -0.421182193 -0.1708188340
[123,]  0.162346955 -0.373532204 -0.1859451456
[124,]  0.154834879 -0.373011508 -0.2368449101
[125,]  0.183910478 -0.348450484 -0.2244149450
[126,]  0.143974653 -0.334954742 -0.2437463746
[127,]  0.114227044 -0.354768723 -0.2144987764
[128,]  0.149624797 -0.320332524 -0.1953442821
[129,]  0.142746746 -0.340139330 -0.1468844460
[130,]  0.152851003 -0.305405043 -0.1351549288
[131,]  0.107049539 -0.316636690 -0.1284494249
[132,]  0.073091599 -0.303829239 -0.1331889813
[133,]  0.107839424 -0.272681990 -0.1368055416
[134,]  0.118283154 -0.252494614 -0.1053063559
[135,]  0.072398494 -0.237755668 -0.0847958532
[136,]  0.059517837 -0.180908806 -0.0905929020
[137,]  0.039553383 -0.180591185 -0.0491303207
[138,]  0.042634942 -0.139181225 -0.0599316430
[139,]  0.016885098 -0.110855863 -0.0461224980
[140,]  0.029367541 -0.066443552 -0.0578010187
[141,]  0.038035060 -0.075493342 -0.0946794576
[142,]  0.012803330 -0.109593922 -0.0993922018
[143,] -0.006112665 -0.092332492 -0.0748384114
[144,] -0.055842772 -0.102661298 -0.0663957471
[145,] -0.053762089 -0.079559592 -0.0306540086
[146,] -0.054564800 -0.092974841  0.0076553080
[147,] -0.109603930 -0.088876953 -0.0111066768
[148,] -0.126195858 -0.065585147 -0.0402903481
[149,] -0.170066266 -0.055912408 -0.0551254378
[150,] -0.176621026 -0.047678410 -0.0085886914
[151,] -0.199271470 -0.067133561  0.0378919079
[152,] -0.217975680 -0.067618035 -0.0087883410
[153,] -0.257456457 -0.049783425 -0.0223842559
[154,] -0.272704358 -0.096452893 -0.0412915843
[155,] -0.316027815 -0.095883899 -0.0163705037
[156,] -0.339895889 -0.105172360 -0.0638253211
[157,] -0.370433422 -0.082652914 -0.0725215549
[158,] -0.404295491 -0.110654160 -0.0979725121
[159,] -0.436810258 -0.114100398 -0.1307936818
[160,] -0.457696429 -0.150728631 -0.1437309521
[161,] -0.489384741 -0.164674621 -0.1305293689
[162,] -0.510568075 -0.126779551 -0.1712092758
[163,] -0.514777169 -0.162419856 -0.2254085454
[164,] -0.525641220 -0.198489273 -0.2083616382
[165,] -0.510951175 -0.233473672 -0.2496678304
[166,] -0.581681525 -0.242111233 -0.2413061036
[167,] -0.560552477 -0.180538050 -0.2677174390
[168,] -0.606585796 -0.189620159 -0.2960730004
[169,] -0.654115526 -0.156899670 -0.3089497966
[170,] -0.576883985 -0.126398826 -0.3241648500
[171,] -0.594242156 -0.154173261 -0.3691733988
[172,] -0.601961347 -0.207570222 -0.3647151729
[173,] -0.533703251 -0.213390945 -0.3465674564
[174,] -0.546383096 -0.173924533 -0.3682669301
[175,] -0.521006928 -0.166768481 -0.4155301474
[176,] -0.511464595 -0.183787596 -0.4833767467
[177,] -0.478901008 -0.207841198 -0.4608020914
[178,] -0.468827351 -0.157772148 -0.4548115352
[179,] -0.465938174 -0.173484652 -0.5211112361
[180,] -0.420236699 -0.170875158 -0.4953664676
[181,] -0.396504192 -0.130697375 -0.4889932512
[182,] -0.447396535 -0.113894063 -0.5064075156
[183,] -0.484425490 -0.092820001 -0.5312820502
[184,] -0.476503950 -0.089267677 -0.4824283134
[185,] -0.459512090 -0.075807327 -0.4386642762
[186,] -0.449193998 -0.035354571 -0.4742995062
[187,] -0.527095273 -0.068042144 -0.4797841709
[188,] -0.556155558 -0.043046155 -0.5099752018
[189,] -0.563498430 -0.014361337 -0.4669530795
[190,] -0.524519636  0.028021695 -0.4687077964
[191,] -0.485789726  0.008630680 -0.4606398695
[192,] -0.484419764  0.013570580 -0.4152098436
[193,] -0.506857523 -0.040822386 -0.4219587288
[194,] -0.531348841 -0.035002061 -0.3963475485
[195,] -0.526607281  0.006830473 -0.3521846310
[196,] -0.500871849  0.102967262 -0.1964042901
[197,] -0.478732639  0.120262173 -0.1626131705
[198,] -0.513526345  0.117612482 -0.1476386160
[199,] -0.539816128  0.150142822 -0.1322040878
[200,] -0.493002262  0.177334468 -0.1242177663
[201,] -0.524986365  0.193104589 -0.0991404724
[202,] -0.511200747  0.224044187 -0.1317841678
[203,] -0.488024712  0.222504801 -0.1577770333
[204,] -0.528216343  0.191451790 -0.1768209470
[205,] -0.566868500  0.196924534 -0.1902350030
[206,] -0.550413817  0.226493587 -0.2254046148
[207,] -0.546837236  0.201567052 -0.2471412797
[208,] -0.563380266  0.154948496 -0.2875227077
[209,] -0.539492142  0.134482059 -0.3195693097
[210,] -0.574184778  0.192609897 -0.3248915048
[211,] -0.586020643  0.161240453 -0.3533905172
[212,] -0.552757265  0.167632860 -0.3744922339
[213,] -0.508608532  0.178062985 -0.3576096442
[214,] -0.476435344  0.193337305 -0.3901713868
[215,] -0.447315589  0.161960436 -0.3605088126
[216,] -0.410334962  0.195639097 -0.3250228600
[217,] -0.400038542  0.162348683 -0.3453517070
[218,] -0.350802589  0.109441053 -0.3100746487
[219,] -0.315430018  0.098952435 -0.3549066169
[220,] -0.267352522  0.120076469 -0.3420753786
[221,] -0.264901975  0.081598183 -0.3210110044
[222,] -0.261839123  0.041329065 -0.3304004296
[223,] -0.235411686  0.003697946 -0.3051234527
[224,] -0.247914423 -0.030718988 -0.2762898272
[225,] -0.210220698 -0.015028929 -0.2600966223
[226,] -0.183643691 -0.031211147 -0.2828954092
[227,] -0.206715296 -0.049419276 -0.3334711763
[228,] -0.200751910 -0.090036279 -0.3109442578
[229,] -0.227990949 -0.085751029 -0.2833921066
[230,] -0.230092653 -0.144166318 -0.3195672151
[231,] -0.252902639 -0.181680249 -0.3050389974
[232,] -0.207905229 -0.200356644 -0.3332903144
[233,] -0.273184255 -0.207782533 -0.3625974546
[234,] -0.252951517 -0.227832286 -0.3263547143
[235,] -0.245263020 -0.266539773 -0.3398503862
[236,] -0.263617282 -0.267687822 -0.2993366991
[237,] -0.271928453 -0.289127276 -0.2738959818
[238,] -0.232810270 -0.317887717 -0.2717614665
[239,] -0.206631948 -0.344206403 -0.2947122199
[240,] -0.202680730 -0.376530216 -0.2703814790
[241,] -0.180219905 -0.353771699 -0.2481614200
[242,] -0.228462519 -0.354378584 -0.2389299792
[243,] -0.213953611 -0.309574029 -0.2221321226
[244,] -0.215787728 -0.303591984 -0.1762191387
[245,] -0.253086950 -0.330343861 -0.1905847271
[246,] -0.242936452 -0.331386602 -0.1384054428
[247,] -0.191427819 -0.344320448 -0.1082002135
[248,] -0.175054527 -0.314657890 -0.0855259406
[249,] -0.210004287 -0.320308953 -0.0819172816
[250,] -0.241189518 -0.350050830 -0.0717729233
[251,] -0.204580797 -0.376590311 -0.0671905813
[252,] -0.202570032 -0.375065145 -0.0146250263
[253,] -0.176187570 -0.355195586 -0.0097494000
[254,] -0.183005220 -0.408394955 -0.0242281078
[255,] -0.156456569 -0.424999156  0.0117492055
[256,] -0.162855973 -0.407820863  0.0483556035
[257,] -0.133260653 -0.435515610  0.0607225362
[258,] -0.142572696 -0.450051625  0.0804876118
[259,] -0.149194672 -0.490913144  0.0992587047
[260,] -0.160585935 -0.531409396  0.1188177440
[261,] -0.126881399 -0.527280653  0.1056633025
[262,] -0.123397467 -0.564678768  0.1450076816
[263,] -0.140449277 -0.593557676  0.1550875136
[264,] -0.100941375 -0.628213042  0.1800089436
[265,] -0.093318898 -0.659064937  0.1595532728
[266,] -0.051852407 -0.633450399  0.1433449868
[267,] -0.078569869 -0.596670194  0.1629182690
[268,] -0.032646856 -0.589794472  0.2083399356
[269,] -0.049494944 -0.545633180  0.2351851737
[270,] -0.041985435 -0.512169900  0.2110412500
[271,] -0.015320271 -0.523767201  0.2328349187
[272,]  0.013546436 -0.514137134  0.2783793628
[273,] -0.023201354 -0.488748142  0.2633339571
[274,] -0.027972306 -0.532388953  0.3102628506
[275,] -0.070948514 -0.505907818  0.3322470707
[276,] -0.091677882 -0.475641021  0.3480388202
[277,] -0.098527627 -0.448603243  0.3852652567
[278,] -0.118154453 -0.407544624  0.4194935147
[279,] -0.144475164 -0.433312281  0.4298096486
[280,] -0.099759570 -0.440562303  0.4511149597
[281,] -0.121979040 -0.480508018  0.4548793915
[282,] -0.121459450 -0.524125180  0.4869732710
[283,] -0.134920726 -0.565971923  0.4725814500
[284,] -0.185971573 -0.566096656  0.4644587717
[285,] -0.212579421 -0.587796590  0.4251558842
[286,] -0.204326154 -0.537663757  0.4092223033
[287,] -0.169750576 -0.526056663  0.4317363127
[288,] -0.144011536 -0.554527950  0.4149690642
[289,] -0.102526020 -0.581369633  0.4083798195
[290,] -0.051764417 -0.606992608  0.4305162260
[291,] -0.013460947 -0.603986052  0.4618526839
[292,]  0.014567247 -0.607554600  0.4262554800
[293,]  0.018544009 -0.566194661  0.4273023069
[294,]  0.042059057 -0.575999524  0.4631951602
[295,]  0.086628693 -0.594428392  0.4655572805
[296,]  0.101887128 -0.546658228  0.4854721033
[297,]  0.128233123 -0.583436789  0.5026387639
[298,]  0.123303950 -0.575678289  0.4495385046
[299,]  0.155767963 -0.545949667  0.4422388095
[300,]  0.176857182 -0.543070378  0.4707788019
[301,]  0.166038329 -0.505679348  0.4685884800
[302,]  0.177122791 -0.494101160  0.4419782966
[303,]  0.251021858 -0.501965535  0.4604050024
[304,]  0.246629154 -0.492202022  0.4139628856
[305,]  0.244902385 -0.461346759  0.4172333164
[306,]  0.296151900 -0.470716347  0.3962177418
[307,]  0.270828137 -0.461319051  0.3637017709
[308,]  0.311475326 -0.434159617  0.3836592242
[309,]  0.330959731 -0.393952906  0.3830414907
[310,]  0.359104036 -0.433925934  0.3592848435
[311,]  0.386488298 -0.435999522  0.3078162245
[312,]  0.396915498 -0.396827605  0.2976860934
[313,]  0.385323403 -0.346067941  0.3037885097
[314,]  0.368395060 -0.348423058  0.2513469944
[315,]  0.417214754 -0.397206786  0.2405133974
[316,]  0.440400832 -0.372693708  0.1990307519
[317,]  0.447171848 -0.412751478  0.1660406746
[318,]  0.462731592 -0.445812585  0.2032113034
[319,]  0.500143290 -0.485512255  0.1850655142
[320,]  0.513758682 -0.485379618  0.2321625861
[321,]  0.517016069 -0.447736977  0.2618442763
[322,]  0.512768720 -0.398642984  0.2487270014
[323,]  0.527922988 -0.410070682  0.1966645682
[324,]  0.570465097 -0.421919300  0.2105770965
[325,]  0.571899255 -0.446518106  0.1639204138
[326,]  0.559831164 -0.407065531  0.1423671062
[327,]  0.530309200 -0.365014861  0.1756271330
[328,]  0.534847586 -0.314758770  0.2118091298
[329,]  0.582503735 -0.296149547  0.1816189163
[330,]  0.615140864 -0.345787707  0.1836304943
[331,]  0.614928936 -0.353279676  0.2470297707
[332,]  0.669725825 -0.390897385  0.1943266211
[333,]  0.643889685 -0.437438001  0.2421168528
[334,]  0.635178453 -0.430797974  0.2925574173
[335,]  0.683995111 -0.387224354  0.2689098102
[336,]  0.654489479 -0.370624415  0.3090913188
[337,]  0.646895633 -0.362879991  0.3535442925
[338,]  0.673147871 -0.327361724  0.3407132974
[339,]  0.697131593 -0.334865097  0.3100557405
[340,]  0.741369096 -0.307489450  0.3537467426
[341,]  0.781325748 -0.271342289  0.3529359467
[342,]  0.803548111 -0.200097674  0.3718931991
[343,]  0.780740379 -0.223849371  0.4267414219
[344,]  0.817858925 -0.263588791  0.4101933508
[345,]  0.864941288 -0.249385837  0.4344557818
[346,]  0.896740144 -0.270950797  0.4659031205
[347,]  0.919624825 -0.250968404  0.4436208633
[348,]  0.912636092 -0.217599169  0.4231405288
[349,]  0.928445599 -0.239479692  0.3926798862
[350,]  0.956531074 -0.221602378  0.3714740005
[351,]  1.233907096 -0.049779643  0.2030086544

[[5]]
[[5]][[1]]
[1] -1.307129 -1.996761




Files produced:
- file 1: 'hap1_full_all_22_100000.txt' (22553 bytes; 44aa39b30ff77a507ec3d93106a31f09)
- file 2: 'hap1_full_all_22_100000beta.txt' (31 bytes; e9ce5676ed46be9d3345594ceaf95299)




R version 3.4.4 (2018-03-15)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04.4 LTS

Matrix products: default
BLAS: /usr/lib/atlas-base/atlas/libblas.so.3.0
LAPACK: /usr/lib/atlas-base/atlas/liblapack.so.3.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8   
 [6] LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8       LC_NAME=C                  LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  base     

other attached packages:
[1] MASS_7.3-49

loaded via a namespace (and not attached):
[1] compiler_3.4.4          tools_3.4.4             matrixStats_0.53.1-9000

Processing time:
   user  system elapsed 
226.560   0.196 226.832 
